pleroma_bot/_utils.py

```diff
@@ -169,7 +171,7 @@ def run(self):
     def join(self):
         if not self.exc:
             self.started_evt.wait()
-        super(PropagatingThread, self).join()
+        super(PropagatingThread, self).join(THREAD_TIMEOUT)
```

Comment:
Simply setting a timeout doesn't make it terminate the thread, stop the execution or even raise a TimeoutError:
https://docs.python.org/3/library/threading.html#threading.Thread.join

It continues the execution and leaves the thread which exceeded the timeout still running, which can be very bad and lead to unexpected behavior if we expect the thread to return something which will be used later on.

You can test this yourself by making the timeout value very small, it won't make any difference because of the spinner process that gets joined.
Maybe the place to implement a timeout and handle it (check if the thread is alive, terminate it, raise an exception, etc.) would be here instead:
https://github.com/robertoszek/pleroma-bot/blob/eb50c6457763e0597e9ddc7d2a20140c474e57b5/pleroma_bot/_utils.py#L58